### DISCUSSION TITLE
is there a way to ignore filetypes?

### DISCUSSION THREAD

**Discussion by danmed** [photoview/photoview#495] [Q&A and Troubleshooting]
I'd like a way to ignore or exclude certain filetypes..

is this currently possible with a docker install?

**Comment by viktorstrate** [comment]
Yes, add a `.photoviewignore` file to your root directory with the following content:

```
*.ext
```

Where `ext` is the file extension of the filetype you want to ignore. You can add multiple lines for different file extensions

**Reply by berwegera** [reply]
Is there also the possibility of a .photoview**include** file to search only certain filetypes `*.jpg,*.jpeg,*.mp4` all other file extensions not?

**Reply by viktorstrate** [reply]
You should be able to do that by writing the following to the `.photoviewignore` file:

```gitignore
# ignore everything
*

# except these file extensions
!*.jpg
!*.jpeg
!*.mp4
```

**Reply by berwegera** [reply]
Perfect!!

**Reply by Europa2010AD** [reply]
> Yes, add a `.photoviewignore` file to your root directory with the following content:
> 
> ```
> *.ext
> ```
> 
> Where `ext` is the file extension of the filetype you want to ignore. You can add multiple lines for different file extensions

Can I somehow use this to tell Photoview to ignore certain folders?

**Reply by viktorstrate** [reply]
@Europa2010AD the `.photoviewignore` file follows the `.gitignore` specification.
See the official documentation on that one for an in-depth description.
https://git-scm.com/docs/gitignore

But in short, yes you can. Just add a line with the path to the album you want to ignore, relative to the ignore file.

**Reply by Europa2010AD** [reply]
Thanks @viktorstrate for the quick reply! To be honest I'm not familiar with .gitignore (I'm no developer / coder), tried to understand the documentation you linked, but it's a little over my head 😅 

But just so I understand it correctly, if I want Photoview to ignore my folders "A" and "B" **inside** the parent folder "Photos", I can place the `.photoviewignore` file in the "Photos" folder (so it's the same level as "A" and "B"), and simply add the line:
```
./A/
./B/
```
Am I correct?

And if folders "A" and "B" were already scanned, do I simply re-scan everything again after adding this `.photoviewignore` file so to get rid of "A" and "B" in the cache?

**Reply by Europa2010AD** [reply]
I can't seem to be able to get this to work -- I tried both of the following in my `.photoviewignore` file:
```
./A/
./B/
```
and
```
/A/
/B/
```
Both folders are still being scanned and showing in my albums. Am I placing the `.photoviewignore` file in the wrong directory? When you say place it in the root directory -- root directory meaning the directory where my photos resides?

**Comment by berwegera** [comment]
Hello together
I don't know if I can help, but I take panoramas consisting of several photos. After I have put the panoramas together, I move the original photos into the subfolder "Panoramagrundlagen". I do not want to index this folder and its contents.

/Photos
/Photos/**.photoviewignore**
/Photos/2020-01-25_Geburtstag_Joel/
/Photos/2020-01-25_Geburtstag_Joel/<Photos-to-Index.jpg>
/Photos/2020-01-25_Geburtstag_Joel/**Panoramagrundlagen/**<Photos-NOT-to-Index.jpg>
/Photos/2020-01-31_Wanderung-Hoernli/
/Photos/2020-01-31_Wanderung-Hoernli/<Photos-to-Index.jpg>
/Photos/2020-01-31_Wanderung-Hoernli/**Panoramagrundlagen/**<Photos-NOT-to-Index.jpg>
....

the .photoviewignore File looks like this:

`Panoramagrundlagen/*`

with
`Panoramagrundlagen/**`

i think not only files also supfolders and files in Subfolders are also ignored

This works for my needs. 
But i think you have to test on your one :)
